src/packages/@ncigdc/routes/Login.js

```diff
@@ -59,112 +67,128 @@ const AWGLoginButton = compose(
           return (window.location.href = '/login?error=no_fence_projects');
         }
       }
-      await dispatch(fetchUser());
-      push({ pathname: '/repository' });
+      const createdSession = await createAWGSession();
```

Comment:
I am looking at this PR from a multi-tab perspective.  (when a user has multiple awg tabs open).  

If this line is executed when a user already has a tab with an active session, then the existing session will get wiped out, due to `removeSessionRenewalTimestamp()`.  But this will only matter if the user failed login from the `openAuthWindow` call above.
1. `openAuthWindow` can reject the promise with 'No cookies enabled'.  
2. `openAuthWindow` can also reject the promise if any errors are thrown inside of it. The only potential error I see is from the use of es6 code in an older browser.  (I haven't looked to see if we use babel to transpile to es5 or polyfill or similar)

Based on these two things, I'd suggest having default behavior at the end of the catch to handle any other errors from auth.   Maybe redirecting to login?

Reply:
Hm though provoking indeed!

Will have to study it a bit more in depth, hopefully over the upcoming RC tag.
One thing I can clear already is that all our code is transpiled/polyfilled correctly on build, so we needn't worry too much about users in older browsers (at least not on that specific subject.)